akka-contrib/src/main/scala/akka/contrib/pattern/ReceivePipeline.scala

```diff
@@ -38,10 +38,10 @@ trait ReceivePipeline extends Actor {
     val around = aroundCache match {
       case Some((`receive`, cached)) ⇒ cached
       case _ ⇒
-        val zipped = pipeline.foldRight[Receive](receive)((outer, inner) ⇒ outer(inner).orElse[Any, Unit](inner))
+        val zipped = pipeline.foldRight[Receive](receive.orElse(PartialFunction(unhandled)))((outer, inner) ⇒ outer(inner).orElse[Any, Unit](inner))
         aroundCache = Some((receive, zipped))
         zipped
     }
-    super.aroundReceive(around, msg)
+    around(msg)
```

Comment:
Indeed. Have re-jigged to use super.aroundReceive as the zero for the fold to pick up the unhandled msg case